### PR TITLE
[srp-server] add API for getting LEASE and KEY-LEASE

### PR DESCRIPTION
**Pull request by wgtdkp**

### CODE REVIEW

File: src/cli/README_SRP_SERVER.md [hunk 1]
```diff
@@ -92,7 +92,29 @@ srp-api-test-0.default.service.arpa.
...
 
-### srp server service
+### lease
+
+Usage: `srp server lease [<min-lease>] [<max-lease>] [<min-key-lease>] [<max-key-lease>]`
+
+Get LEASE and KEY-LEASE values.
+
+```bash
+> srp server lease
```
Comment (simonlingoogle) [inline src/cli/README_SRP_SERVER.md:102]: implement in OTCI?

File: include/openthread/srp_server.h [hunk 2]
```diff
@@ -112,6 +112,22 @@ otError otSrpServerSetDomain(otInstance *aInstance, const char *aDomain);
  */
 void otSrpServerSetEnabled(otInstance *aInstance, bool aEnabled);
 
+/**
+ * This method returns LEASE & KEY-LEASE range from the SRP server.
```
Comment (abtink) [inline include/openthread/srp_server.h:116]:
```suggestion
 * This function returns LEASE & KEY-LEASE range from the SRP server.
```

File: include/openthread/srp_server.h [hunk 3]
```diff
@@ -112,6 +112,22 @@ otError otSrpServerSetDomain(otInstance *aInstance, const char *aDomain);
...
+ * @param[out]  aMinKeyLease  A pointer to the minimum KEY-LEASE interval in seconds.
+ * @param[out]  aMaxKeyLease  A pointer to the maximum KEY-LEASE interval in seconds.
+ *
+ */
+void otSrpServerGetLeaseRange(otInstance *aInstance,
+                              uint32_t   *aMinLease,
+                              uint32_t   *aMaxLease,
+                              uint32_t   *aMinKeyLease,
+                              uint32_t   *aMaxKeyLease);
+
```
Comment (abtink) [inline include/openthread/srp_server.h:130]: I guess `make-pretty` here.

**Review by abtink — APPROVED** [review]
LGTM. 👍 
One small suggestion below.

File: tools/otci/otci/otci.py [hunk 4]
```diff
@@ -947,6 +947,11 @@ def __parse_srp_server_service_txt(self, txt: str) -> Dict[str, Union[bytes, boo
 
         return txt_dict
 
+    def srp_server_get_lease(self):
```
Comment (simonlingoogle) [inline tools/otci/otci/otci.py:950]:
```suggestion
    def srp_server_get_lease(self) -> Tuple[int, int, int, int]:
```

We tend to use typing in OTCI.

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

File: include/openthread/srp_server.h [hunk 5]
```diff
@@ -113,7 +113,23 @@ otError otSrpServerSetDomain(otInstance *aInstance, const char *aDomain);
...
+ * @param[out]  aMaxLease     A pointer to the maximum LEASE interval in seconds.
+ * @param[out]  aMinKeyLease  A pointer to the minimum KEY-LEASE interval in seconds.
+ * @param[out]  aMaxKeyLease  A pointer to the maximum KEY-LEASE interval in seconds.
+ *
+ */
+void otSrpServerGetLeaseRange(otInstance *aInstance,
+                              uint32_t *  aMinLease,
+                              uint32_t *  aMaxLease,
+                              uint32_t *  aMinKeyLease,
+                              uint32_t *  aMaxKeyLease);
```
Comment (jwhui) [inline include/openthread/srp_server.h:129]: Rather than having a parameter for each value, what about defining a struct that is passed as a single param?

Comment (wgtdkp) [reply include/openthread/srp_server.h]: @jwhui Udpated to use a `otSrpServerLeaseConfig` struct. Also defined an internal `Srp::Server::LeaseConfig` class with convenient utility methods (e..g `IsValid`, `GrantLease` and `GrantKeyLease`).

File: include/openthread/srp_server.h [hunk 6]
```diff
@@ -74,7 +74,19 @@ typedef void otSrpServerService;
...
- * This method returns the domain authorized to the SRP server.
+ * This structure includes SRP server LEASE and KEY-LEASE configurations.
+ *
+ */
+typedef struct otSrpServerLeaseConfig
+{
+    uint32_t mMinLease;    // The minimum LEASE interval in seconds.
+    uint32_t mMaxLease;    // The maximum LEASE interval in seconds.
+    uint32_t mMinKeyLease; // The minimum KEY-LEASE interval in seconds.
+    uint32_t mMaxKeyLease; // The maximum KEY-LEASE interval in seconds.
```
Comment (jwhui) [inline include/openthread/srp_server.h:85]:
```suggestion
    uint32_t mMinLease;    ///< The minimum LEASE interval in seconds.
    uint32_t mMaxLease;    ///< The maximum LEASE interval in seconds.
    uint32_t mMinKeyLease; ///< The minimum KEY-LEASE interval in seconds.
    uint32_t mMaxKeyLease; ///< The maximum KEY-LEASE interval in seconds.
```

**Review by abtink — APPROVED** [review]
Looks great. Thanks.

File: src/core/net/srp_server.cpp [hunk 7]
```diff
@@ -128,41 +124,62 @@ void Server::SetEnabled(bool aEnabled)
...
 
-uint32_t Server::GrantKeyLease(uint32_t aKeyLease) const
+uint32_t Server::LeaseConfig::GrantKeyLease(uint32_t aKeyLease) const
 {
     OT_ASSERT(mMinKeyLease <= mMaxKeyLease);
 
     return (aKeyLease == 0) ? 0 : OT_MAX(mMinKeyLease, OT_MIN(mMaxKeyLease, aKeyLease));
 }
 
+void Server::GetLeaseConfig(LeaseConfig &aLeaseConfig)
```
Comment (jwhui) [inline src/core/net/srp_server.cpp:167]:
```suggestion
void Server::GetLeaseConfig(LeaseConfig &aLeaseConfig) const
```

File: src/core/net/srp_server.hpp [hunk 8]
```diff
@@ -469,22 +490,27 @@ class Server : public InstanceLocator, private NonCopyable
     void SetEnabled(bool aEnabled);
 
     /**
-     * This method sets LEASE & KEY-LEASE range that is acceptable by the SRP server.
+     * This method returns the LEASE and KEY-LEASE configurations.
+     *
+     * @param[out]  aLeaseConfig  A reference to the `LeaseConfig` instance.
+     *
+     */
+    void GetLeaseConfig(LeaseConfig &aLeaseConfig);
```
Comment (jwhui) [inline src/core/net/srp_server.hpp:498]:
```suggestion
    void GetLeaseConfig(LeaseConfig &aLeaseConfig) const;
```

**Review by jwhui — APPROVED** [review]

**Pull request merged**